tests/components/sensor/test_tcp.py

```diff
@@ -2,198 +2,133 @@
     def test_name_not_set(self, mock_update):
         """Should return the superclass name property if not set in config."""
         config = copy(TEST_CONFIG)
         del config[tcp.CONF_NAME]
         entity = Entity()
-        sensor = tcp.Sensor(self.hass, config)
+        sensor = tcp.TcpSensor(self.hass, config)
         assert sensor.name == entity.name
 
-    @patch('homeassistant.components.sensor.tcp.Sensor.update')
+    @patch('homeassistant.components.sensor.tcp.TcpSensor.update')
     def test_state(self, mock_update):
-        """Should return the contents of _state."""
-        sensor = tcp.Sensor(self.hass, TEST_CONFIG)
+        """Check the contents of _state."""
+        sensor = tcp.TcpSensor(self.hass, TEST_CONFIG)
         uuid = str(uuid4())
         sensor._state = uuid
         assert sensor.state == uuid
 
-    @patch('homeassistant.components.sensor.tcp.Sensor.update')
+    @patch('homeassistant.components.sensor.tcp.TcpSensor.update')
     def test_unit_of_measurement(self, mock_update):
-        """Should return the configured unit of measurement."""
-        sensor = tcp.Sensor(self.hass, TEST_CONFIG)
-        assert sensor.unit_of_measurement == TEST_CONFIG[tcp.CONF_UNIT]
+        """Check the configured unit of measurement."""
+        sensor = tcp.TcpSensor(self.hass, TEST_CONFIG)
+        assert sensor.unit_of_measurement == \
+            TEST_CONFIG[tcp.CONF_UNIT_OF_MEASUREMENT]
 
-    @patch("homeassistant.components.sensor.tcp.Sensor.update")
+    @patch("homeassistant.components.sensor.tcp.TcpSensor.update")
     def test_config_valid_keys(self, *args):
-        """Should store valid keys in _config."""
-        sensor = tcp.Sensor(self.hass, TEST_CONFIG)
+        """Store valid keys in _config."""
+        sensor = tcp.TcpSensor(self.hass, TEST_CONFIG)
         for key in TEST_CONFIG:
             assert key in sensor._config
 
-    def test_validate_config_valid_keys(self):
-        """Should return True when provided with the correct keys."""
-        assert tcp.Sensor.validate_config(TEST_CONFIG)
-
-    @patch("homeassistant.components.sensor.tcp.Sensor.update")
-    def test_config_invalid_keys(self, mock_update):
-        """Shouldn't store invalid keys in _config."""
-        config = copy(TEST_CONFIG)
-        config.update({
-            "a": "test_a",
-            "b": "test_b",
-            "c": "test_c"
-        })
-        sensor = tcp.Sensor(self.hass, config)
-        for invalid_key in "abc":
-            assert invalid_key not in sensor._config
-
-    def test_validate_config_invalid_keys(self):
-        """Test with invalid keys plus some extra."""
-        config = copy(TEST_CONFIG)
-        config.update({
-            "a": "test_a",
-            "b": "test_b",
-            "c": "test_c"
-        })
-        assert tcp.Sensor.validate_config(config)
-
-    @patch("homeassistant.components.sensor.tcp.Sensor.update")
-    def test_config_uses_defaults(self, mock_update):
```

Comment:
This seems like a useful test (so do most others that got removed, just to lazy to comment on them all).